cmd/agent/subcommands/processchecks/command_test.go

```diff
@@ -11,34 +11,44 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	"go.uber.org/fx"
 
 	"github.com/DataDog/datadog-agent/cmd/agent/command"
 	"github.com/DataDog/datadog-agent/cmd/process-agent/subcommands/check"
-	ipcmock "github.com/DataDog/datadog-agent/comp/core/ipc/mock"
+	"github.com/DataDog/datadog-agent/comp/core"
+	configComponent "github.com/DataDog/datadog-agent/comp/core/config"
+	ipc "github.com/DataDog/datadog-agent/comp/core/ipc/def"
+	ipcfx "github.com/DataDog/datadog-agent/comp/core/ipc/fx"
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
 func TestCommand(t *testing.T) {
-	fxutil.TestOneShotSubcommand(t,
-		Commands(newGlobalParamsTest(t)),
-		[]string{"processchecks", "process"},
-		check.RunCheckCmd,
-		func(_ *check.CliParams) {},
-	)
-}
-
-func newGlobalParamsTest(t *testing.T) *command.GlobalParams {
 	testDir := t.TempDir()
 
 	configPath := path.Join(testDir, "datadog.yaml")
-
-	// creating in-memory auth artifacts
-	ipcmock.New(t)
-
 	err := os.WriteFile(configPath, []byte("hostname: test"), 0644)
 	require.NoError(t, err)
 
-	return &command.GlobalParams{
-		ConfFilePath: configPath,
-	}
+	// Check command should work when an Agent is running, so we need to
+	// ensure we have exisiting IPC auth artifacts.
+	// This is done by building the IPC component
+	// with the `ipcfx.ModuleReadWrite()` module.
+	fxutil.Test[ipc.Component](t,
+		ipcfx.ModuleReadWrite(),
+		core.MockBundle(),
+		fx.Replace(configComponent.MockParams{
+			Params: configComponent.Params{
+				ConfFilePath: configPath,
+			},
+		}),
+	)
```

Comment:
We're trying to move away from the MockParams as they have a lot of issues.

Could we update the code with:
```suggestion
    configComponent.NewMockFromYAMLFile(t, configPath)

	// Check command should work when an Agent is running, so we need to
	// ensure we have exisiting IPC auth artifacts.
	// This is done by building the IPC component
	// with the `ipcfx.ModuleReadWrite()` module.
	fxutil.Test[ipc.Component](t,
		ipcfx.ModuleReadWrite(),
		core.MockBundle(),
	)
```

The config mock is design to be created multiple times per check without issue (we don't reset the mock if it's already mocked, this allow multiple helpers to be combined).

Reply:
Good to know ! 
Done in 77db77a314ed2961a19df9edef74151c2c4246d4.
(I also made the change in `cmd/agent/subcommands/processchecks/command_test.go`)